src/pages/blocks/[blockId]/index.page.tsx

```diff
@@ -40,9 +46,19 @@ export default function BlockDetails(
 
 function BlockHeading({
   block,
+  mappedEvmBlockNum,
 }: InferGetServerSidePropsType<typeof getServerSideProps>): JSX.Element {
+  const network = useNetwork().connection;
+  const [metachainBlockUrl, setMetachainBlockUrl] = useState<string>();
+
+  if (mappedEvmBlockNum) {
+    const txUrl = getMetaScanBlockUrl(network, mappedEvmBlockNum);
+    if (!metachainBlockUrl) {
+      setMetachainBlockUrl(txUrl);
+    }
+  }
```

Comment:
useMemo would be a better approach imo since I don't forsee the logic for computing metachainBlockUrl be complex in the future, 

++ simplified code without the need for separate state management.

haven't tested the code below, but something along these lines
```suggestion
const metachainBlockUrl = useMemo(() => {
  if (mappedEvmBlockNum) {
    return getMetaScanBlockUrl(network, mappedEvmBlockNum);
  }
  return undefined;
}, [mappedEvmBlockNum, network]);
```